contracts/src/Gateway.sol

```diff
@@ -265,12 +268,21 @@ contract Gateway is IGateway, IInitializable {
             revert InvalidAgentExecutionPayload();
         }
 
-        bytes memory call = abi.encodeCall(AgentExecutor.execute, params.payload);
+        (AgentExecuteCommand command, bytes memory payload) = abi.decode(params.payload, (AgentExecuteCommand, bytes));
+
+        bytes memory call = abi.encodeCall(AgentExecutor.execute, (params.agentID, command, payload));
 
         (bool success, bytes memory returndata) = Agent(payable(agent)).invoke(AGENT_EXECUTOR, call);
         if (!success) {
             revert AgentExecutionFailed(returndata);
         }
+
+        if (command == AgentExecuteCommand.RegisterToken) {
```

Comment:
I think the approach I suggested to return a command from from `Agent.invoke` doesn't look as neat as the Agent calling `Gateway.registerToken`.

So I think we should go back to that old way.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/fddb9cf3b38ee6c25e4f23fc26daa70e0b9ea73e